File: gateway/internal/model/conversation.go
package model

import (
	"encoding/json"
	"time"
)

// Conversation represents a chat session owned by a user.
type Conversation struct {
	ID          uint64          `gorm:"column:id;primaryKey;autoIncrement"`
	UUID        string          `gorm:"column:uuid;type:char(36);uniqueIndex:uk_conversations_uuid;not null"`
	UserID      uint64          `gorm:"column:user_id;type:bigint unsigned;not null;index:idx_conversations_user_id"`
	Title       string          `gorm:"column:title;type:varchar(255)"`
	ResourceIDs json.RawMessage `gorm:"column:resource_ids;type:json"`
	ModelConfig json.RawMessage `gorm:"column:model_config;type:json"`
	CreatedAt   time.Time       `gorm:"column:created_at;type:datetime;autoCreateTime"`
	UpdatedAt   time.Time       `gorm:"column:updated_at;type:datetime;autoUpdateTime"`

	User User `gorm:"foreignKey:UserID;references:ID;constraint:OnDelete:CASCADE"`
}

File: client/lib/domain/entities/conversation.dart
/// A conversation record.
class Conversation {
  const Conversation({
    required this.id,
    this.title = '',
    this.resourceIds = const <String>[],
    this.modelConfig,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String title;
  final List<String> resourceIds;
  final Map<String, dynamic>? modelConfig;
  final DateTime createdAt;
  final DateTime updatedAt;

  Conversation copyWith({
    String? id,
    String? title,
    List<String>? resourceIds,
    Map<String, dynamic>? modelConfig,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Conversation(
      id: id ?? this.id,
      title: title ?? this.title,
      resourceIds: resourceIds ?? this.resourceIds,
      modelConfig: modelConfig ?? this.modelConfig,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
